src/dev/lupio/lupio_rtc.hh

```diff
@@ -0,0 +1,59 @@
+#ifndef __DEV_LUPIO_LUPIO_RTC_HH__
+#define __DEV_LUPIO_LUPIO_RTC_HH__
+
+#include "base/bitunion.hh"
+#include "base/logging.hh"
+#include "debug/LupioRTC.hh"
+#include "dev/io_device.hh"
+#include "params/LupioRTC.hh"
+#include "sim/core.hh"
+#include "sim/system.hh"
+
+namespace gem5
+{
+
+using namespace RiscvISA;
+
+/** Real-Time Clock (LUPIO_RTC) */
+class LupioRTC : public BasicPioDevice
+{
+
+  protected:
+    System *system;
+    const struct tm time;
+
+    private:
+
+    static const int LUPIO_RTC_SECD = 0x00;
+    static const int LUPIO_RTC_MINT = 0x01;
+    static const int LUPIO_RTC_HOUR = 0x02;
+    static const int LUPIO_RTC_DYMO = 0x03;
+    static const int LUPIO_RTC_MNTH = 0x04;
+    static const int LUPIO_RTC_YEAR = 0x05;
+    static const int LUPIO_RTC_CENT = 0x06;
+    static const int LUPIO_RTC_DYWK = 0x07;
+    static const int LUPIO_RTC_DYYR = 0x08;
+    static const int LUPIO_RTC_MAX = 0x09;
+
+  public:
+    PARAMS(LupioRTC);
+    LupioRTC(const Params &params);
+    virtual ~LupioRTC();
+
+    /**
+    * Function to return current time and date using system's wall-clock time
+    */
+
```

Comment:
Remove blank line between comment and function protype? (check with coding convention in gem5)